### PR TITLE
bump NN to 127.1.0, Maps to 10.12.0-beta.1, common to 23.4.0-beta.1

### PR DESCRIPTION
**Pull request by dzinad**

### CODE REVIEW

File: changelog/unreleased/bugfixes/6986.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+- Improved duration calculation for EV routes to account for charging time.
+- Added snapping feature to the navigation for walking and cycling profiles.
+- Parse route distance and duration to correspond to eV routes ???
```
Comment (Zayankovsky) [inline changelog/unreleased/bugfixes/6986.md:3]: What is going on here?

File: changelog/unreleased/bugfixes/6986.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+- Started invalidating passed alternatives conditionally on the fork structure.
+- Started using offline router if server returns 404 or 403.
+- Delayed discarding alternatives until MapMatching algorithm is confident enough.
+- Fixed a possible native crash in `ElectronicHorizonObserver` on application exit.
+- Fixed redundant memory usage for routing algorithm in Valhalla.
+- Fixed inconsistent choice between re-route and switching to alternative due to tha fact that the alternative was not checked on every off-route state.
```
Comment (Zayankovsky) [inline changelog/unreleased/bugfixes/6986.md:9]:
```suggestion
- Fixed inconsistent choice between re-route and switching to alternative due to the fact that the alternative was not checked on every off-route state. 
```

**Review by Zayankovsky — APPROVED** [review]

File: changelog/unreleased/bugfixes/6986.md (hunk 1 above)
Comment (dzinad) [reply changelog/unreleased/bugfixes/6986.md:3]: Forgot to remove this line, thanks.

File: changelog/unreleased/bugfixes/6986.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+- Improved duration calculation for EV routes to account for charging time.
+- Added snapping feature to the navigation for walking and cycling profiles.
```
Comment (VysotskiVadim) [inline changelog/unreleased/bugfixes/6986.md:2]:
```suggestion
```
Is that related to native reroute controller? If yes, we probably don't need it.

File: changelog/unreleased/bugfixes/6986.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+- Started invalidating passed alternatives conditionally on the fork structure.
+- Started using offline router if server returns 404 or 403.
+- Delayed discarding alternatives until MapMatching algorithm is confident enough.
```
Comment (VysotskiVadim) [inline changelog/unreleased/bugfixes/6986.md:5]:
```suggestion
- Increased the distance at which the navigator discards passed alternative route in `NavigationRouteAlternativesObserver#onRouteAlternatives`. This reduces the chance of discarding alternative routes to which a driver deviated from the primary route.
```
I have already described this feature in this way here: https://github.com/mapbox/mapbox-navigation-android/pull/6966/files

But maybe it's okay to have different descriptions for different versions if they're both great 🙂

File: changelog/unreleased/bugfixes/6986.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+- Fixed a possible native crash in `ElectronicHorizonObserver` on application exit.
+- Fixed redundant memory usage for routing algorithm in Valhalla.
```
Comment (VysotskiVadim) [inline changelog/unreleased/bugfixes/6986.md:7]:
```suggestion
- Fixed redundant memory usage for onboard router.
```
Maybe?

**Review by VysotskiVadim — APPROVED** [review]

File: changelog/unreleased/bugfixes/6986.md (hunk 3 above)
Comment (dzinad) [reply changelog/unreleased/bugfixes/6986.md:2]: It's customer-related, I think we should leave it.
See `mapbox-navigation-native/pull/6741`.
